django/db/models/fields/related.py

```diff
@@ -615,6 +615,12 @@ def _check_unique_target(self):
         if not self.foreign_related_fields:
             return []
 
+        # If a model defines Meta.primary_key and a foreign key refers to it,
+        # the check should be skipped (since primary keys are unique).
+        pk = self.remote_field.model._meta.primary_key
+        if pk and set(pk) == {field.attname for field in self.foreign_related_fields}:
+            return []
+
```

Comment:
I think we should eventually extend `Options.total_unique_constraints` instead or replace with an a method that returns set of fields that are totally unique together by combining `field.unique`, `Meta.unique_together`, `Meta.constraints` (for unique constraints), and now `primary_key`.

Reply:
Hmm, and also `field.primary_key` I suppose? I would lean towards replacing it with a new method then, since changing this `total_unique_constraints` would break backwards compatibility (although it's not really a public API, is it?).

Reply:
I have replaced this code and introduced a new property `Options.pk_fields` - this will achieve the same thing but it's a little bit more in line with what you suggested.